core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -585,7 +585,7 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
     Inode<?> lastInode = currentInodeDirectory.getChild(name);
     if (lastInode != null) {
       // Lock the last inode before subsequent operations, and add it to the lock group.
-      lockList.lockWrite(lastInode);
+      lockList.lockWriteAndCheckFullPath(lastInode, currentInodeDirectory, name);
```

Comment:
Its based on the caller's needs. If they want to ensure a particular inode is being accessed, then they should check the full path. If they are iterating over children, only the parent needs to be checked.